hogben/tests/test_samples.py

```diff
@@ -0,0 +1,313 @@
+    """
+    sample = request.getfixturevalue(sample_class)
+    q, r = sample._get_reflectivity_profile(0.005, 0.4, 500, 1, 1e-7, 2)
+    assert len(q) == len(r)
+    assert len(q) > 0 # Make sure array is not empty
+
+def test_to_refl1d_instance(refnx_sample):
+    """
+    Tests whether a conversion from refnx to a refl1d structure correctly
+    results in a structure of the refl1d instance.
+    """
+    refnx_sample.to_refl1d()
+    assert isinstance(refnx_sample.structure, refl1d.model.Stack)
+
+def test_to_refnx_instance(refl1d_sample):
+    """
+    Tests whether a conversion from refl1d to a refnx structure correctly
+    results in a structure of the refnx instance.
+    """
+    refl1d_sample.to_refnx()
+    assert isinstance(refl1d_sample.structure, refnx.reflect.Structure)
+
+def test_to_refl1d_values(refnx_sample):
+    """
+    Tests whether the strucutral parameters are correctly carried over when
+    converting from a refnx sample to a refl1d sample.
+    """
+    refl1d_sample = copy.deepcopy(refnx_sample)
+    refl1d_sample.to_refl1d()
+    assert compare_sample_structure(refl1d_sample, refnx_sample)
+
+def test_to_refnx_values(refl1d_sample):
+    """
+    Tests whether the structural parameters are correctly carried over when
+    converting from a refl1d sample to a refnx sample.
+    """
+    refnx_sample = copy.deepcopy(refl1d_sample)
+    refnx_sample.to_refnx()
+    assert compare_sample_structure(refl1d_sample, refnx_sample)
+
+def test_simple_sample():
+    """
+    Tests whether simple_sample leads to a valid refnx structure that can be
+    converted to refl1d
+    """
+    simple_sample = samples.simple_sample()
+    simple_sample.to_refl1d()
+
+def test_many_param_sample():
+    """
+    Tests whether many_param_sample leads to a valid refnx structure that
+    can be converted to refl1d
+    """
+    many_param_sample = samples.many_param_sample()
+    many_param_sample.to_refl1d()
+
+def test_thin_layer_sample_1():
+    """
+    Tests whether thin_layer_sample_1 leads to a valid refnx structure that
+    can be converted to refl1d
+    """
+    thin_layer_sample_1 = samples.thin_layer_sample_1()
+    thin_layer_sample_1.to_refl1d()
+
+def test_thin_layer_sample_2():
+    """
+    Tests whether thin_layer_sample_2 leads to a valid refnx structure that
+    can be converted to refl1d
+    """
+    thin_layer_sample_2 = samples.thin_layer_sample_2()
+    thin_layer_sample_2.to_refl1d()
+
+def test_similar_sld_sample_1():
+    """
+    Tests whether similar_sld_sample_1 leads to a valid refnx structure that
+    can be converted to refl1d
+    """
+    similar_sld_sample_1 = samples.similar_sld_sample_1()
+    similar_sld_sample_1.to_refl1d()
+
+def test_similar_sld_sample_2():
+    """
+    Tests whether similar_sld_sample_2 leads to a valid refnx structure that
+    can be converted to refl1d
+    """
+    similar_sld_sample_2 = samples.similar_sld_sample_2()
+    similar_sld_sample_2.to_refl1d()
```

Comment:
I'm not sure the tests for all of the samples are needed, but if you want to include them, then I think you need to have some sort of check in here, as you run the `to_refl1d()` function, but there are ways where it could fail silently I think.

Reply:
I've parameterized them, as it's basically just the same test anyway and this makes this a lot cleaner. It also now checks if the converted sample is actually a refl1d sample instance, so that it doesn't silently become a None type for instance.

In principle these are not the most important tests in the world. As it doesn't really test any functions that are needed. But it doesn't hurt to check that the predefined samples are still working I guess.